src/app/kuma/components/ApplicationShell.vue

```diff
@@ -0,0 +1,280 @@
+          name="notifications"
+        >
+          <KAlert
+            v-if="!can('use state')"
+            class="mb-4"
+            appearance="warning"
+          >
+            <template #alertMessage>
+              <ul>
+                <!-- eslint-disable vue/no-v-html  -->
+                <li
+                  data-testid="warning-GLOBAL_STORE_TYPE_MEMORY"
+                  v-html="t('common.warnings.GLOBAL_STORE_TYPE_MEMORY')"
+                />
+                <!-- eslint-enable -->
+              </ul>
+            </template>
+          </KAlert>
+          <OnboardingNotification />
+        </slot>
+        <slot name="default" />
+      </div>
+    </div>
+  </div>
+</template>
+<script lang="ts" setup>
+import { KUI_ICON_SIZE_30 } from '@kong/design-tokens'
+import { CogIcon, HelpIcon } from '@kong/icons'
+import GithubButton from 'vue-github-button'
+
+import { useEnv, useI18n, useCan } from '@/app/application'
+import UpgradeCheck from '@/app/common/UpgradeCheck.vue'
+import OnboardingNotification from '@/app/onboarding/components/OnboardingNotification.vue'
+
+const env = useEnv()
+const can = useCan()
+const { t } = useI18n()
+
+</script>
+<style lang="scss">
+html.is-fullscreen {
+  .application-shell > header {
+    display: none;
+  }
+  .application-shell > .app-content-container > nav {
+    display: none;
+  }
+  .application-shell > .app-content-container > .app-notifications {
+    display: none;
+  }
+}
+html.no-navigation {
+  .application-shell > .app-content-container > nav {
+    display: none;
+  }
+}
+</style>
+
+<style lang="scss" scoped>
+.app-content-container {
+  padding-top: var(--AppHeaderHeight, initial);
+  display: var(--AppDisplay);
+  grid-template-columns: var(--AppSidebarWidth) 1fr;
+}
+
+.app-main-content {
+  padding: var(--AppContentPadding);
+}
+
+header {
+  position: fixed;
+  z-index: 11;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: var(--AppHeaderHeight);
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  gap: $kui-space-80;
+  padding-right: $kui-space-80;
+  padding-left: $kui-space-80;
+  border-bottom: $kui-border-width-10 solid $kui-color-border;
+  background-color: $kui-color-background;
+}
+
+.horizontal-list {
+  display: flex;
+  align-items: center;
+  gap: $kui-space-80;
+}
+// This wrapping element is necessary. It ensures that the sidebar can participate in a grid or flex container.
+nav {
+  position: static;
+}
+
+nav ul {
+  list-style-type: none;
+  padding: 0;
+  width: var(--AppSidebarWidth);
+  position: fixed;
+  z-index: 10;
+  top: var(--AppHeaderHeight);
+  bottom: 0;
+  left: 0;
+  overflow-y: auto;
+  padding-top: $kui-space-40;
+  padding-right: $kui-space-40;
+  border-right: $kui-border-width-10 solid $kui-color-border;
+  background-color:  $kui-color-background;
+}
+nav > ul > :is(li) {
+  margin-left: $kui-space-40;
+}
+nav > ul > :is(li) + :is(li) {
+  margin-top: $kui-space-20;
+}
+nav > ul > :is(li) > :is(a) {
```

Comment:
**Suggestion**: Please use class names on elements you want to select. There’s no need to tie selectors to the HTML structure unless that is useful to enforce (not the case here).

Something like 

```suggestion
.site-nav :deep(.nav-item) {
```

would be much more robust.

In fact, this type of style illustrates how having a dedicated NavItem component was a good idea: it can own its styles instead of relying on a separate component to own them.

Reply:
I added an `AppNavigator` component that wraps the slot in a `li` with a class name on it and now use that to internally wrap all the `Navigator`s. I then use that class from ApplicationShell to apply the styles.

Reply:
Sounds good 👍🏻